std/typecons.d

```diff
@@ -288,6 +288,25 @@ private:
     assert(!uf2.isEmpty);
 }
 
+unittest
+{
+    struct Bar {int val;}
+    struct Foo
+    {
+        Unique!Bar bar = new Bar;
+    }
+
+    Foo foo;
+    foo.bar.val = 6;
+    const Foo* ptr = &foo;
+    static assert(is(typeof(ptr) == const(Foo*)));
```

Comment:
These static asserts seem a bit redundant; they follow from the fact that const is transitive in D. You can infer this from the assignment of `ptr` compiling.